ooniapi/services/ooniprobe/src/ooniprobe/routers/v1/probe_services.py

```diff
@@ -590,3 +565,234 @@ def random_web_test_helpers(th_list: List[str]) -> List[Dict]:
+@router.post("/sign_credential", tags=["anonymous_credentials"])
+def sign_credential(register_request: RegisterRequest, session : PostgresSessionDep):
+
+    manifest = OONIProbeManifest.get_by_version(session, register_request.manifest_version)
+    if manifest is None:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND,
+            {
+                "error" : "manifest_not_found",
+                "message" : f"No manifest with version '{register_request.manifest_version}' was found"}
+            )
+    state: OONIProbeServerState = manifest.server_state
+    protocol_state = state.to_protocol()
+
+    try:
+        resp = protocol_state.handle_registration_request(register_request.credential_sign_request)
+    except (ProtocolError, CredentialError, DeserializationFailed) as e:
+        raise to_http_exception(e)
+
+    return RegisterResponse(
+        credential_sign_response=resp,
+        emission_day=protocol_state.today()
+    )
+
+
+def to_http_exception(error: ProtocolError | CredentialError | DeserializationFailed):
+
+    type_to_str = {
+        ProtocolError : "protocol_error",
+        DeserializationFailed : "deserialization_failed",
+        CredentialError : "credential_error"
+    }
+    type_str = type_to_str[type(error)]
+
+    assert isinstance(error, (ProtocolError, CredentialError, DeserializationFailed))
+    status_code = status.HTTP_400_BAD_REQUEST if isinstance(error, DeserializationFailed) else status.HTTP_403_FORBIDDEN
+
+    return HTTPException(
+        status_code=status_code,
+        detail={
+            "error" : type_str,
+            "message" : str(error)
+        }
+    )
+
+class SubmitMeasurementRequest(BaseModel):
+
+    format: str
+    content: Dict[str, Any]
+    # -- < Anonymous Credentials > ----------------------
+    # not post quantum, in the future we might want to use a hashed key for storage
+    nym: str
+    zkp_request: str
+    age_range: Tuple[int,int]
+    msm_range: Tuple[int,int]
+    manifest_version: str
+
+class SubmitMeasurementResponse(BaseModel):
+    """
+    Acknowledge
+    """
+
+    measurement_uid: str | None = Field(
+        examples=["20210208220710.181572_MA_ndt_7888edc7748936bf"], default=None
+    )
+    is_verified: bool = Field(
+        description="if the ZKP was able to verify this request"
+    )
+    submit_response: str | None = Field(
+        description="Anonymous credential verification response. Null if verification failed"
+    )
+
+@router.post("/submit_measurement/{report_id}")
+async def submit_measurement(
+    report_id: str,
+    request: Request,
+    submit_request: SubmitMeasurementRequest,
+    response: Response,
+    cc_reader: CCReaderDep,
+    asn_reader: ASNReaderDep,
+    settings: SettingsDep,
+    s3_client: S3ClientDep,
+    session: PostgresSessionDep,
+    content_encoding: str = Header(default=None),
+) -> SubmitMeasurementResponse | Dict[str, Any]:
+    """
+    Submit measurement
```

Comment:
Can you write a better docstring?

Also, it would be good to know where we plug in the values for "valid" `probe_age_range`s and `probe_msm_range` (can you rename these variables to that)